test/log-iterator.spec.js

```diff
@@ -0,0 +1,243 @@
+      })
+
+      it('iterates with lte and explicit amount', async () => {
+        let amount = 10
+
+        let it = log1.iterator({
+          lte: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        assert.strictEqual([...it].length, 10)
+
+        let i = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+        }
+      })
+
+      it('iterates with lt and explicit amount', async () => {
+        let amount = 10
+
+        let it = log1.iterator({
+          lt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 1
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      it('iterates with gt and explicit amount', async () => {
+        let amount = 5
+
+        let it = log1.iterator({
+          gt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (72 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      it('iterates with gte and explicit amount', async () => {
+        let amount = 12
+
+        let it = log1.iterator({
+          gt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL',
+          amount: amount
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (79 - i++))
+          count++
+        }
+        assert.strictEqual(count, amount)
+      })
+
+      /* eslint-disable camelcase */
+      it('iterates with combinations of lt, gt, lte, and gte', async () => {
+        // lt + gt
+        let it_lt_gt = log1.iterator({
+          gt: 'zdpuAvZ6cDrY57iDxEQ5iSXibAeJPcBFD1jHnAmKq1RUiyuoK',
+          lt: 'zdpuAuPfNkZmPdMxbyw1MxEwuv3XdrMMtFaTSi7sgc7j8VnzW'
+        })
+        let it_lt_gt_array = [...it_lt_gt]
+
+        // neither hash should appear in the array
+        assert.strictEqual(it_lt_gt_array.map(e => e.hash)
+          .indexOf('zdpuAvZ6cDrY57iDxEQ5iSXibAeJPcBFD1jHnAmKq1RUiyuoK'), -1)
+        assert.strictEqual(it_lt_gt_array.map(e => e.hash)
+          .indexOf('zdpuAuPfNkZmPdMxbyw1MxEwuv3XdrMMtFaTSi7sgc7j8VnzW'), -1)
+        assert.strictEqual(it_lt_gt_array.length, 10)
+
+        // lt + gte
+        let it_lt_gte = log1.iterator({
+          gte: 'zdpuAyw2rPe5ML6YmYe7DEyAq3CbWH3QpWQ4P6F4ZFroEebzz',
+          lt: 'zdpuAxv8bUaFNmCp51zN3AwCLy83FfVCcp7GeTwCdtpVddziP'
+        })
+        let it_lt_gte_array = [...it_lt_gte]
+
+        // neither hash should appear in the array
+        assert.strictEqual(it_lt_gte_array.map(e => e.hash)
```

Comment:
The mapping here and below could be done in one place before the asserts. Saves a few cycles :)